src/ng/compile.js

```diff
@@ -2670,7 +2670,9 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
      */
     function addDirective(tDirectives, name, location, maxPriority, ignoreDirective, startAttrName,
                           endAttrName) {
-      if (name === ignoreDirective) return null;
+      if (typeof ignoreDirective === 'string' && name === ignoreDirective) return null;
+      if (typeof ignoreDirective === 'function' && !ignoreDirective.call(null, name)) return null;
```

Comment:
I thought we might like a value for "this" but I wanted to discuss with the community about the best option before choosing.
As of now just `ignoreDirective(name)` is perfectly fine.

Also I realized exposing the full parameters list to the function might be a good idea, like:

`if (typeof ignoreDirective === 'function' && !ignoreDirective(tDirectives, name, location, maxPriority, ignoreDirective, startAttrName, endAttrName)) return null;`

Someone might need also other parameters like "location" in case he wants to process only the class directives or the attribute ones.